aws/resource_aws_network_acl_association.go

```diff
@@ -0,0 +1,140 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNetworkAclAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNetworkAclAssociationCreate,
+		Read:   resourceAwsNetworkAclAssociationRead,
+		Update: resourceAwsNetworkAclAssociationUpdate,
+		Delete: resourceAwsNetworkAclAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"subnet_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"network_acl_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsNetworkAclAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	naclId := d.Get("network_acl_id").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	log.Printf(
+		"[INFO] Creating network acl association: %s => %s",
+		subnetId,
+		naclId)
+
+	association, err_association := findNetworkAclAssociation(subnetId, conn)
```

Comment:
Please prefer camelCase-d variables instead of snake-case-d ones, like `errAssociation`